src/atlas/projection/detail/StretchProjection.cc

```diff
@@ -0,0 +1,384 @@
+                 }
+             }
+
+             lamphi = point;
+             lamphi = (lamphi >= 360) ? lamphi -360.0 : lamphi ;
+          }
+
+
+          /**
+          * SECTION 5: Reset lat/lons
+          * If longitude domain crosses the meridian (0 degrees) then
+          * add 360 degrees to points East of the meridian
+          */
+          if (L_long){
+              lamphi = (lamphi < 180) ? lamphi + 360.0 : lamphi ;
+          }
+
+       } 
+       else {
+           /*
+            * SECTION 6: regular grids
+            * var_ratio = 1.0
+            */
+            lamphi = point;
+            if (L_long){
+                lamphi = (lamphi < 180) ? lamphi + 360.0 : lamphi ;
+            }
+        }
+    return lamphi;
+
+}
+
+///< xy unstretched, only unrotation
+template <typename Rotation>
+void StretchLAM<Rotation>::lonlat2xy( double crd[] ) const {
+
+    //unrotate
+    rotation_.rotate( crd );
+
+    // PUT the unstretch, I don't have it nows
+
+}
+
+///< From unstretched to stretched
+template <typename Rotation>
+void StretchLAM<Rotation>::xy2lonlat( double crd[] ) const {
+
+    /** PUT here the stretch for a point that come input
+    * give number of power as input,work it out using as from the start of regular grid.
+    * atlas::PointXY unstretchedXY = crd;
+    * atlas::PointXY stretchedXY = stretch_LAM_gen(unstretchedXY);
+    * stretch_LAM_gen(crd[]);
+    */
+
+    double lam_hires_size; ///< size regular grid x
+    double phi_hires_size; ///< size regular grid y
+    double lambda_start; ///< start grid x
+    double phi_start; ///< start grid y
+    double add_xf_; ///< distance end of the grid and internal regular grid
+    double add_yf_; ///< distance end of the grid and internal regular grid
+    double check_x; ///< check middle of the previous regular grid
+    double check_y; ///< check middle of the previous regular grid
+    double epsilon = 0.01; ///< use in check if the same value
+    double epstest = 0.00000000001; ///< correction used to change from double to integer
+    int nx_, ny_;
+    int n_stretchedx_, n_stretchedy_, n_x_rim_, n_y_rim_;
+
+    ///< original domain size includes the points for the rim
+    double deltax_all = (endx_ - startx_);
+    double deltay_all = (endy_ - starty_);
+
+    if (var_ratio_ == 1) {
+           lam_hires_size = deltax_all;
+           phi_hires_size = deltay_all;
+           lambda_start = x_reg_start_;
+           phi_start = y_reg_start_;
+
+         } else {
+              lam_hires_size = x_reg_end_ - x_reg_start_;
+              phi_hires_size = y_reg_end_ - y_reg_start_;
+
+              /** check for start of the new regular LAM grid x and y
+              * in the middle of the previous regular grid
+              */
+
+              add_xf_ = (deltax_all + epstest - lam_hires_size) / 2.;
+              add_yf_ = (deltay_all + epstest - phi_hires_size) / 2.;
```

Comment:
A bit nitpicky but you can avoid accidentially having uninitialised data if you move the declaration of variables as close as possible to their initialisation. So for example here you could (I think) declare these two variables in these lines rather than the beginning of the function.

Reply:
Now, I defined nx_ and ny_ before the if loop. I can't define deltax_all  inside the else statement.

Reply:
Sorry I wasn't clear, I was thinking something along the lines of:
```suggestion
              double add_yf_ = (deltay_all + epstest - phi_hires_size) / 2.;
```
And something similar for `add_xf_`. You could then I think remove the declarations on lines 301 and 302. At least I think that is right! I haven't actually don't it myself.